src/sw.ts

```diff
@@ -23,7 +23,16 @@ function urlsToEntries(urls: string[], hash: string): PrecacheEntry[] {
   });
 }
 
-export function setupPwa() {
+/**
+ * Add PWA capabilities.
+ *
+ * **WARNING**: "prompt" mode not available yet.
+ *
+ * @param mode
+ * @default "auto-update"
+ */
+export function setupPwa(mode: "auto-update" | "prompt" = "auto-update") {
```

Comment:
what's the difference?

Reply:
with auto-update, when a new sw detected, once new sw activated the page will be reloaded, the new sw will control all clients

with prompt, the user should show a prompt for update, and once the user click on the reload button, the new sw will control all clients; for example, if the user is filling a form, the prompt strategy will allow the user to save the forma data before installing the new version

Reply:
makes sense, thanks.